src/screens/AirdropDashboard/AirdropDashboardScreen.tsx

```diff
@@ -1,38 +1,107 @@
 import { CompositeNavigationProp } from '@react-navigation/native';
 import { StackNavigationProp } from '@react-navigation/stack';
-import React, { FC } from 'react';
-import { View, StyleSheet } from 'react-native';
+import { differenceBy } from 'lodash';
+import React, { FC, useEffect } from 'react';
+import { View, StyleSheet, ActivityIndicator } from 'react-native';
+import { connect } from 'react-redux';
 
 import { Header, ScreenTemplate } from 'app/components';
-import { Route, RootStackParams } from 'app/consts';
+import { Route, RootStackParams, Wallet, WalletPayload } from 'app/consts';
 import { isAfterAirdrop } from 'app/helpers/airdrop';
+import { walletToAddressesGenerationBase } from 'app/helpers/wallets';
+import { ApplicationState } from 'app/state';
+import {
+  checkSubscription,
+  CheckSubscriptionAction,
+  subscribeWallet,
+  SubscribeWalletAction,
+} from 'app/state/airdrop/actions';
+import * as airdropSelectors from 'app/state/airdrop/selectors';
 
-import { AirdropFinished, AirdropInProgress } from './components';
+import { AirdropFinished, AirdropInProgress, Footer } from './components';
 
 const i18n = require('../../../loc');
 
 interface Props {
+  wallets: Wallet[];
+  error: boolean;
+  isLoading: boolean;
+  isInitialized: boolean;
+  checkSubscription: (wallets: Wallet[]) => CheckSubscriptionAction;
   navigation: CompositeNavigationProp<
     StackNavigationProp<RootStackParams, Route.MainTabStackNavigator>,
     StackNavigationProp<RootStackParams, Route.AirdropDashboard>
   >;
+  subscribedWallets: Wallet[];
+  subscribeWallet: (payload: { wallet: WalletPayload; id: string }) => SubscribeWalletAction;
 }
 
-// TODO: when implementing data fetching, add loading and error screens here.
-// loading: https://app.zeplin.io/project/5f0c8686b8151e82242dd409/screen/6005bdd8de723b5789229476
-// error: https://app.zeplin.io/project/5f0c8686b8151e82242dd409/screen/6005a32229c91b1a89551702
+export const AirdropDashboardScreen: FC<Props> = ({
+  navigation,
+  isInitialized,
+  wallets,
+  checkSubscription,
+  subscribedWallets,
+  subscribeWallet,
+  isLoading,
+  error,
+}) => {
+  useEffect(() => {
+    checkSubscription(wallets);
+  }, [checkSubscription, wallets]);
+  const airdropFinished = isAfterAirdrop();
 
-export const AirdropDashboardScreen: FC<Props> = ({ navigation }) => (
-  <ScreenTemplate header={<Header isBackArrow title={i18n.airdrop.title} />}>
-    <View style={styles.wrapper}>
-      {isAfterAirdrop() ? <AirdropFinished navigation={navigation} /> : <AirdropInProgress navigation={navigation} />}
+  const getAvailableWallets = (wallets: Wallet[]) => differenceBy(wallets, subscribedWallets, 'id');
+
+  return !isInitialized ? (
```

Comment:
is it possible that the user will be on this screen without loaded wallets?

Reply:
no it s not, thanks